### DISCUSSION TITLE
Building stlink under macOS

### DISCUSSION THREAD

**Discussion by chriskuku** [stlink-org/stlink#1237] [Q&A (resolved)] [answered]
In my desperation of not getting stlink v 1.6.1 working under macOS (always getting 2022-03-31T12:24:49 WARN: unknown chip id! 0x374b) I'm trying to build stlink.

Intuitively I did a git clone, cd'd to stlink and did a "make".

As a result I got binaries in /Users/me/Downloads/stlink/build/Release/bin.
Starting gets :
```$ ./st-info --probe
/usr/local/stlink/chips: No such file or directory
Failed to enter SWD mode
Found 1 stlink programmers
  version:    V2J39S27
  serial:     066CFF485153826687123124
  flash:      0 (pagesize: 0)
  sram:       0
  chipid:     0x000
```

Or:
```
$ ./st-flash erase
st-flash 1.7.0-186-gc4762e6
/usr/local/stlink/chips: No such file or directory
Failed to enter SWD mode
Failed to connect to target
$ 
```
Why do I get /usr/local/stlink/chips: No such file or directory ?

**Comment by gszy** [comment, marked as answer]
> Intuitively I did a git clone, cd'd to stlink and did a "make". As a result I got binaries in /Users/me/Downloads/stlink/build/Release/bin. […] Why do I get /usr/local/stlink/chips: No such file or directory ?

I haven’t tried it, but `make install` should copy the missing chip config files.

**Reply by chriskuku** [reply]
Thanks. Overlooked that. It did it !

**Reply by Nightwalker-87** [reply]
This step was indeed missing in the specific section for macOS in our compiling tutorial.
It'll be updated soon. Thanks for the note!
